src/textual/widgets/_digit_display.py

```diff
@@ -0,0 +1,271 @@
+from textual.app import ComposeResult
+from textual.reactive import reactive
+from textual.widget import Widget
+from textual.widgets import Static
+
+
+_character_map: dict[str, str] = {}
+
+# in the mappings below, we use underscores to make spaces more visible,
+# we will strip them out later
+_character_map[
+    "0"
+] = """
+┏━┓
+┃╱┃
+┗━┛
+"""
+
+_character_map[
+    "1"
+] = """
+·┓·
+ ┃·
+╺┻╸
+"""
+
+_character_map[
+    "2"
+] = """
+╺━┓
+┏━┛
+┗━╸
+"""
+
+_character_map[
+    "3"
+] = """
+╺━┓
+ ━┫
+╺━┛
+"""
+
+_character_map[
+    "4"
+] = """
+╻ ╻
+┗━┫
+  ╹
+"""
+
+_character_map[
+    "5"
+] = """
+┏━╸
+┗━┓
+╺━┛
+"""
+
+_character_map[
+    "6"
+] = """
+┏━╸
+┣━┓
+┗━┛
+"""
+
+_character_map[
+    "7"
+] = """
+╺━┓
+  ╹
+ ╹·
+"""
+
+_character_map[
+    "8"
+] = """
+┏━┓
+┣━┫
+┗━┛
+"""
+
+_character_map[
+    "9"
+] = """
+┏━┓
+┗━┫
+╺━┛
+"""
+
+_character_map[
+    " "
+] = """
+···
+···
+···
+"""
+
+
+_character_map[
+    "X"
+] = """
+╻ ╻
+·╋·
+╹ ╹
+"""
+
+_character_map[
+    "Y"
+] = """
+╻ ╻
+·┳·
+·╹·
+"""
+
+_character_map[
+    "Z"
+] = """
+╺━┓
+·▞·
+┗━╸
+"""
+
+
+_character_map[
+    "."
+] = """
+··
+··
+·•
+"""
+
+
+_character_map[
+    ","
+] = """
+··
+··
+·▞
+"""
+
+_character_map[
+    "+"
+] = """
+···
+╺╋╸
+···
+"""
+
+_character_map[
+    "-"
+] = """
+···
+╺━╸
+···
+"""
+
+_character_map[
+    "="
+] = """
+···
+╺━·
+╺━·
+"""
+
+_character_map[
+    "*"
+] = """
+···
+·✱·
+···
+"""
+
+
+_character_map[
+    "/"
+] = """
+··╻
+·▞·
+╹··
+"""
+
+_character_map[
+    "^"
+] = """
+·╻·
+▝·▘
+···
+"""
+
+
+_VIRTUAL_SPACE = "·"
```

Comment:
Consider moving this up, next to the comment about stripping an auxiliary character, and fix the comment that mentions underscores when we're actually using this dot (which is a good idea, by the way).